rllib/agents/trainer.py

```diff
@@ -1048,15 +1050,24 @@ def make_callbacks():
 
     @staticmethod
     def _validate_config(config: PartialTrainerConfigDict):
-        if config.get("_use_trajectory_view_api") and \
-                config.get("framework") != "torch":
+        use_traj_view_api = config.get("_use_trajectory_view_api")
+        model_is_auto_lstm = config.get("model", {}).get("use_lstm")
+        # Enable trajectory view API iff None and framework=torch and
+        # model=auto-LSTM (meaning it's a time_major model).
+        if use_traj_view_api is None:
+            if config.get("framework") == "torch" and model_is_auto_lstm:
+                config["_use_trajectory_view_api"] = True
+            else:
+                config["_use_trajectory_view_api"] = False
```

Comment:
I think we should only flip to to False programmatically. For setting it to True, that should come from the agent config. That way it's pretty clear: it's on by default for certain algos, except when it's not supported (e.g., TensorFlow).

Reply:
Also, since we want to enable it for non-LSTM case (which is the more important case). I would remove this auto lstm check. Basically, I am thinking to replace the changes in this function in favor of:

```
if config[use_traj_view] and framework != torch:
   config[use_traj_view] = False   # disable, not supported

if not config[use_traj_view] and auto_lstm:
   config[model][_time_major] = False   # disable, not supported (maybe log a message)
```

That way, everything is done with config, and we only are safeguarding unsupported cases.